frontend/src/views/Chat.vue

```diff
@@ -1,13 +1,263 @@
+
+const toggleSelectView = () => {
+    hide.value = !hide.value;
+}
+
+// Variabile per controllare lo stato della funzione di copia del prompt
+const isCopying = ref(false);
+
+const copyToClipboard = (event) => {
+    // Interrompo se c'è già una copia in corso
+    if (isCopying.value) return; 
+    isCopying.value = true;
+    // Risalgo al messaggio più vicino al bottone cliccato
+    const messageContent = event.currentTarget.closest('.message').querySelector('p');
+    if (!messageContent) return;
+    // Estraggo il contenuto dall'elemento ed elimino eventuali spazi all'inizio e alla fine della stringa
+    const text = messageContent.textContent.trim();
+    navigator.clipboard.writeText(text)
+        .then(() => {
+            console.log('Text copied to clipboard: ', text);
+            messageSuccess('Copy', 'Text copied to clipboard');
+            setTimeout(() => {
+                isCopying.value = false;
+            }, 2000);
+        })
+        .catch((err) => {
+            console.error('Error when copying to clipboard: ', err);
+            messageError('Copy', 'Error when copying to clipboard');
+        });
+};
+
+// Ritorno il nome del dizionario dati selezionato
+const getDictionaryName = (code) => {
+  const dict = dictionaries.value.find(dict => dict.code === code);
+  return dict ? dict.name + ' (.json)' : 'Choose a dictionary';
+};
+
 </script>
 <template>
-    <FloatLabel>
-    <InputText id="username" v-model="value" />
-    <label for="username">Scrivi prompt</label>
-</FloatLabel>
+
+    <div id="chat" class="flex flex-column justify-between">
+        <!-- TITLE -->
+        <div id="titlebar-container" class="card p-3">
+            <div id="chat-title" class="flex flex-row align-items-center">
+                <!--<h1 ref="chatTitle" class="m-1 text-xl font-semibold">Choose a dictionary</h1>-->
+                <h1 class="m-1 text-xl font-semibold">{{ getDictionaryName(selectedDictionary) }}</h1>
+                <ToggleButton v-model="checked" onLabel="Show" offLabel="Hide" onIcon="pi pi-check" offIcon="pi pi-times" class="w-9rem m-1" aria-label="Hide or Show" @click="toggleSelectView"/>
+            </div>
+            <Divider :class="{hide: hide}" />
+            <div :class="{hide: hide}" class="flex flex-wrap flex-row mt-2">
+                <Dropdown filter v-model="selectedDictionary" :options="dictionaries" optionLabel="name"
+                        optionValue="code" placeholder="Choose dictionary..." class="w-fit max-w-12rem md:max-w-max h-fit m-2" />
+                <Button severity="secondary" icon="pi pi-info" rounded class="m-2" />
+    
+                <Dropdown v-model="selectedDbms" :options="dbms" optionLabel="name" optionValue="code"
+                    class="w-fit h-fit m-2" />
+                <Dropdown v-model="selectedLanguage" :options="languages" optionLabel="name" optionValue="code"
+                    class="w-fit h-fit m-2" />
+            </div>
+        </div>
```

Comment:
Siccome i filtri occupano tanto posto se proviamo a lasciarli sempre visibili sulla destra in alto? vi metto come spunto la stessa intestazione dei dizionari)

```suggestion
        <div class="flex justify-content-between flex-wrap">
            <div class="flex m-2">
                <h1>Chat</h1>
            </div>
            <div class="flex m-2">
                <InputGroup >
                    <FloatLabel >
                        <Dropdown filter v-model="selectedDictionary" :options="dictionaries" optionLabel="name"
                            optionValue="code" placeholder="Choose dictionary..."  inputId="in-dictionary" class="w-fit h-fit m-2 mr-0"/>
                        <label for="indictionary">Dictionary</label>
                    </FloatLabel>

                    <Button severity="info" icon="pi pi-info" class="h-fit m-2 ml-0" />
                </InputGroup>

                <FloatLabel>
                    <Dropdown v-model="selectedDbms" :options="dbms" optionLabel="name" optionValue="code"
                    class="w-fit h-fit m-2" inputId="in-dbms" />
                    <label for="in-dbms">DBMS</label>
                </FloatLabel>

                <FloatLabel>
                    <Dropdown v-model="selectedLanguage" :options="languages" optionLabel="name" optionValue="code"
                    class="w-fit h-fit m-2" inputId="in-language"/>
                    <label for="in-language">Language</label>
                </FloatLabel>
            </div>
        </div>
```

Reply:
Questo è un attimo da valutare.
Di sicuro l'input group va implementato per il dizionario dati. Però non so se lasciare tutti gli input sempre visibili, soprattutto per schermi piccoli. Secondo me è corretto dare all'utente la possibilità di nasconderli, e metterli dentro una card. Però magari si può gestire un po' meglio lo spazio.